common/classes/com/arjuna/common/internal/util/propertyservice/BeanPopulator.java

```diff
@@ -29,8 +29,12 @@ public class BeanPopulator
 {
     private static final ConcurrentMap<String, Object> beanInstances = new ConcurrentHashMap<String, Object>();
 
+    private static ConcurrentMap<String, Object> getBeanInstances() {
+        return beanInstances;
+    }
```

Comment:
@mmusgrov It should only introduce `getBeanInstances` method in `narayana` and I think we can use `@Substitute` in Quarkus extension to rewrite this method to check the `null` value.

In `narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/graal/PropertiesFactorySubstitution.java`
```java
    @Substitute
    private static ConcurrentMap<String, Object> getBeanInstances() {
        if (beanInstances == null) {
            synchronized (BeanPopulator.class) {
                if (beanInstances == null) {
                    beanInstances = new ConcurrentHashMap<String, Object>();
                }
            }
        }
        return beanInstances;
    }

```
Does it make sense and could be accepeted by Narayana?